In [1]:
import pandas as pd

In [2]:
df = pd.read_csv("AB_NYC_2019.csv")

In [3]:
df.head()

id                                              name  host_id  \
0  2539                Clean & quiet apt home by the park     2787   
1  2595                             Skylit Midtown Castle     2845   
2  3647               THE VILLAGE OF HARLEM....NEW YORK !     4632   
3  3831                   Cozy Entire Floor of Brownstone     4869   
4  5022  Entire Apt: Spacious Studio/Loft by central park     7192   

     host_name neighbourhood_group neighbourhood  latitude  longitude  \
0         John            Brooklyn    Kensington  40.64749  -73.97237   
1     Jennifer           Manhattan       Midtown  40.75362  -73.98377   
2    Elisabeth           Manhattan        Harlem  40.80902  -73.94190   
3  LisaRoxanne            Brooklyn  Clinton Hill  40.68514  -73.95976   
4        Laura           Manhattan   East Harlem  40.79851  -73.94399   

         room_type  price  minimum_nights  number_of_reviews last_review  \
0     Private room    149               1                  9  19-10-2018   
1  Entire home/apt    225               1                 45  21-05-2019   
2     Private room    150               3                  0         NaN   
3  Entire home/apt     89               1                270  05-07-2019   
4  Entire home/apt     80              10                  9  19-11-2018   

   reviews_per_month  calculated_host_listings_count  availability_365  
0               0.21                               6               365  
1               0.38                               2               355  
2                NaN                               1               365  
3               4.64                               1               194  
4               0.10                               1                 0

In [4]:
df["id"]=df["id"].astype(str)
df["host_id"]=df["host_id"].astype(str)
df["latitude"]=df["latitude"].astype(str)
df["longitude"]=df["longitude"].astype(str)

How does the data look mathematically?

In [5]:
df.describe()

price  minimum_nights  number_of_reviews  reviews_per_month  \
count  48906.000000    48906.000000       48906.000000       38854.000000   
mean     152.711324        7.031612          23.300454           1.373151   
std      240.128713       20.512489          44.607175           1.680270   
min        0.000000        1.000000           0.000000           0.010000   
25%       69.000000        1.000000           1.000000           0.190000   
50%      106.000000        3.000000           5.000000           0.720000   
75%      175.000000        5.000000          24.000000           2.020000   
max    10000.000000     1250.000000         629.000000          58.500000   

       calculated_host_listings_count  availability_365  
count                    48906.000000      48906.000000  
mean                         7.142702        112.782031  
std                         32.948926        131.620370  
min                          1.000000          0.000000  
25%                          1.000000          0.000000  
50%                          1.000000         45.000000  
75%                          2.000000        227.000000  
max                        327.000000        365.000000

range of minimum nights for listings is 1 and 1250

# Categorical Data

In [6]:
df.nunique()

id                                48895
name                              47896
host_id                           37457
host_name                         11452
neighbourhood_group                   5
neighbourhood                       221
latitude                          19048
longitude                         14718
room_type                             3
price                               674
minimum_nights                      109
number_of_reviews                   394
last_review                        1764
reviews_per_month                   937
calculated_host_listings_count       47
availability_365                    366
dtype: int64

In [7]:
df.columns

Index(['id', 'name', 'host_id', 'host_name', 'neighbourhood_group',
       'neighbourhood', 'latitude', 'longitude', 'room_type', 'price',
       'minimum_nights', 'number_of_reviews', 'last_review',
       'reviews_per_month', 'calculated_host_listings_count',
       'availability_365'],
      dtype='object')

In [8]:
df["room_type"].value_counts()

Entire home/apt    25414
Private room       22332
Shared room         1160
Name: room_type, dtype: int64

In [9]:
df["room_type"].value_counts(normalize = True)

Entire home/apt    0.519650
Private room       0.456631
Shared room        0.023719
Name: room_type, dtype: float64

In [10]:
df["neighbourhood_group"].value_counts()

Manhattan        21669
Brooklyn         20107
Queens            5666
Bronx             1091
Staten Island      373
Name: neighbourhood_group, dtype: int64

# Numerical Data

In [11]:
df["price"].value_counts(bins = 5)

(-10.001, 2000.0]    48820
(2000.0, 4000.0]        54
(4000.0, 6000.0]        16
(6000.0, 8000.0]         9
(8000.0, 10000.0]        7
Name: price, dtype: int64

In [12]:
bins = [-10,0, 50,100, 200,500,800,2000,4000,10000]
df["price"].value_counts(bins = bins)

(50.0, 100.0]        17373
(100.0, 200.0]       16588
(200.0, 500.0]        7340
(0.0, 50.0]           6550
(500.0, 800.0]         624
(800.0, 2000.0]        334
(2000.0, 4000.0]        54
(4000.0, 10000.0]       32
(-10.001, 0.0]          11
Name: price, dtype: int64

It is mainly helpful in small datasets.

## Measures of central tendency

In [13]:
df["price"].mean()

152.71132376395533

In [14]:
df["price"].median()

106.0

In [15]:
df["price"].std()

240.1287131622509

In [16]:
df["minimum_nights"].mean()

7.031611663190611

In [17]:
df["minimum_nights"].median()

3.0

## Measure of Spread

In [18]:
df["price"].skew()

19.120831694826197

In [19]:
df["price"].kurt()

585.7930484394186

How many listings have availability throughout the year (365 days)

In [20]:
df[df["availability_365"]==365].shape[0]

1295

In [21]:
df.corr()

C:\Users\GFG19189\AppData\Local\Temp\ipykernel_1056\1134722465.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  df.corr()


price  minimum_nights  number_of_reviews  \
price                           1.000000        0.042771          -0.048014   
minimum_nights                  0.042771        1.000000          -0.080093   
number_of_reviews              -0.048014       -0.080093           1.000000   
reviews_per_month              -0.030608       -0.121772           0.549291   
calculated_host_listings_count  0.057478        0.127917          -0.072375   
availability_365                0.081817        0.144146           0.172002   

                                reviews_per_month  \
price                                   -0.030608   
minimum_nights                          -0.121772   
number_of_reviews                        0.549291   
reviews_per_month                        1.000000   
calculated_host_listings_count          -0.009414   
availability_365                         0.185818   

                                calculated_host_listings_count  \
price                                                 0.057478   
minimum_nights                                        0.127917   
number_of_reviews                                    -0.072375   
reviews_per_month                                    -0.009414   
calculated_host_listings_count                        1.000000   
availability_365                                      0.225680   

                                availability_365  
price                                   0.081817  
minimum_nights                          0.144146  
number_of_reviews                       0.172002  
reviews_per_month                       0.185818  
calculated_host_listings_count          0.225680  
availability_365                        1.000000